keps/sig-storage/1979-object-storage-support/README.md

```diff
@@ -3,703 +3,742 @@
-	+ return the credentials to the workload owning the `BucketAccessRequest`.
-+  Existing User and New Bucket: In this scenario, we have the user account created in the backend storage system, but the account is not associated to the `Bucket`.
-	+ add the access privileges to the `Bucket`.
-	+ return the credentials to the workload owning the `BucketAccessRequest`.
-+  Existing User and existing Bucket: In this scenario, the user account has access policy defined on the `Bucket`.  The existing user privileges in the backend may conflict with the privileges that the user is requesting.
-	+ FAIL, if existing access policy is different from the requested policy.
-	+ if the existing privileges match the requested privileges, return the credentials to the workload owning the `BucketAccessRequest`.
-+ A Service Account specified and the cloud platform identity integration maps Kubernetes ServiceAccounts to the account in the backend storage system. No need to create credentials here.
-	
-Upon success, the `BucketAccess` instance is ready and the app workload can access backend storage.
+Since COSI automates the lifecycle management of credentials used for accessing buckets, and also standardizes a pattern for consuming object storage buckets in kubernetes, it is desirable to access existing buckets in a similar fashion to a bucket created by COSI. This user story explains the steps to re-use a bucket:
 
-### Static Provisioning
-"Driverless" allows the existing workloads to make use of COSI without the need for vendors to create drivers. The following steps show the details of the workflow:
-+ Admin creates `Bucket` instance that references an existing backend bucket.
-+ User creates a `BucketRequest` (BR) that references this `Bucket`.
-+ User creates `BucketAccessRequest` (BAR) that references their BR, a `BucketAccessClass`, and their secret containing access tokens.
-+ COSI detects the existence of the BAR and follows the [grant access](#grant-bucket-access) steps.
-+ COSI detects the existence of the BAR, BA and executes a validation process.
-+ As a validation step, COSI ensures that the `BucketRequest.bucketInstanceName` matches the `BucketAccessRequest.bucketAccessName.bucketInstanceName`. In other words, the two related `Bucket` references match, meaning that the BR and BAR->BA point to the same `Bucket`.
+The following stakeholders are involved in this lifecycle:
 
----
+ - Admin - creates the Bucket API object
+ - User - specifies bucket in their pod definition 
 
-# gRPC Definitions
+###### 1. Admin creates a Bucket API object
 
-There is one service defined by COSI - `Provisioner`. This will need to be satisfied by the vendor-provisioner in order to be COSI-compatible
+A Bucket with `bucketID` indicates that this Bucket has already been created outside of COSI. This bucket API object is not created as a result of a BucketRequest processed by COSI, it is rather created by the admin. 
```

Comment:
In the normal flow, who deletes a Bucket and when?  If I delete my BR, does thew Bucket go away or does it leak forever?

Then who is responsible for deleting these "manual" buckets?  How do I prevent them leaking or being GC'ed prematurely?

Reply:
I'd love to see the full lifecycle explained

Reply:
I'm working on this, along with service account based auth. Gimme an hour or so

Reply:
> Then who is responsible for deleting these "manual" buckets? How do I prevent them leaking or being GC'ed prematurely?

The admin has to delete these buckets. It is a manual operation, the admin is expected to ensure that two Bucket API objects do not point to the same bucket in the OSP.